FifthLesson/VectorProgram/VectorProgram/Model/Vector.cs

```diff
@@ -1,115 +1,137 @@
-            result.CoordinateSecond += vector2.CoordinateSecond;
-            result.CoordinateThird += vector2.CoordinateThird;
+            result.FirstCoordinate += second.FirstCoordinate;
+            result.SecondCoordinate += second.SecondCoordinate;
+            result.ThirdCoordinate += second.ThirdCoordinate;
 
             return result;
         }
 
-        public static Vector operator -(Vector vector1, Vector vector2)
+        public static Vector operator -(Vector first, Vector second)
         {
-            var result = new Vector(vector1);
+            var result = new Vector(first);
 
-            result.CoordinateFirst -= vector2.CoordinateFirst;
-            result.CoordinateSecond -= vector2.CoordinateSecond;
-            result.CoordinateThird -= vector2.CoordinateThird;
+            result.FirstCoordinate -= second.FirstCoordinate;
+            result.SecondCoordinate -= second.SecondCoordinate;
+            result.ThirdCoordinate -= second.ThirdCoordinate;
 
             return result;
         }
 
         public static Vector operator *(double numeric, Vector vector)
         {
-            return new Vector(vector.CoordinateFirst * numeric, vector.CoordinateSecond * numeric, vector.CoordinateThird * numeric);
+            return new Vector(vector.FirstCoordinate * numeric, vector.SecondCoordinate * numeric, vector.ThirdCoordinate * numeric);
         }
 
         public static Vector operator *(Vector vector, double numeric)
         {
             return numeric * vector;
         }
 
-        public static Vector operator *(Vector vector1, Vector vector2)
+        public static Vector operator *(Vector first, Vector second)
         {
-            return new Vector(vector1.CoordinateFirst * vector2.CoordinateFirst, vector1.CoordinateSecond * vector2.CoordinateSecond, vector1.CoordinateThird * vector2.CoordinateThird);
+            return new Vector(first.FirstCoordinate * second.FirstCoordinate, first.SecondCoordinate * second.SecondCoordinate, first.ThirdCoordinate * second.ThirdCoordinate);
         }
 
-        public static bool operator ==(Vector vector1, Vector vector2)
-        {
-            var result = (vector1.CoordinateFirst == vector2.CoordinateFirst && vector1.CoordinateSecond == vector2.CoordinateSecond && vector1.CoordinateThird == vector2.CoordinateThird);
-            return result;
-        }
+        public static bool operator ==(Vector first, Vector second) => Equals(first, second);
+
+        public static bool operator !=(Vector first, Vector second) => !Equals(first, second);
+        #endregion
 
-        public static bool operator !=(Vector vector1, Vector vector2)
+        public virtual bool Equals(Vector other)
         {
-            return !(vector1 == vector2);
+            if ((object)this == (object)other) return true;
+            if ((object)other == null) return false;
+            if (GetType() != other.GetType()) return false;
```

Comment:
What for you try to compare types of these objects? Parameter has type **Vector**.

Reply:
Yes exactly. Removed